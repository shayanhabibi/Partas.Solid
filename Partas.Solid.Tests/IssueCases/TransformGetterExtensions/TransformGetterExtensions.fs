module Partas.Solid.Tests.IssueCases.TransformGetterExtensions

open Partas.Solid
open Partas.Solid.TanStack.Table
open Fable.Core

[<Erase>]
type TransformGetterExtensions() =
    inherit VoidNode()
    [<SolidTypeComponentAttribute>]
    member props.constructor =
        let table = createTable<int>(TableOptions())
        div() {
            Show(when' = unbox (table.getRowModel().rows.Length), fallback = (div())) {
                For(each = table.getRowModel().rows) {
                    yield fun row index ->
                        div().data("state", unbox (row.getIsSelected() && unbox "selected")) {
                            For(each = row.getVisibleCells()) {
                                yield fun cell index ->
                                    div() { flexRender(cell.column.columnDef.cell, cell.getContext) }
                            }
                        }
                }
            }
        }