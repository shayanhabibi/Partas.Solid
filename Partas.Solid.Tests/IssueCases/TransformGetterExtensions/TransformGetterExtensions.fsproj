<Project Sdk="Microsoft.NET.Sdk">

    <PropertyGroup>
        <OutputType>Exe</OutputType>
        <TargetFramework>net9.0</TargetFramework>
    </PropertyGroup>

    <ItemGroup>
        <Compile Include="Program.fs" />
        <Compile Include="Table.fs" />
        <Compile Include="ColumnFiltering.fs" />
        <Compile Include="ColumnFaceting.fs" />
        <Compile Include="ColumnOrdering.fs" />
        <Compile Include="ColumnPinning.fs" />
        <Compile Include="ColumnSizing.fs" />
        <Compile Include="ColumnVisibility.fs" />
        <Compile Include="GlobalFaceting.fs" />
        <Compile Include="GlobalFiltering.fs" />
        <Compile Include="Sorting.fs" />
        <Compile Include="Grouping.fs" />
        <Compile Include="Expanding.fs" />
        <Compile Include="Pagination.fs" />
        <Compile Include="RowPinning.fs" />
        <Compile Include="RowSelection.fs" />
        <Compile Include="TransformGetterExtensions.fs" />
    </ItemGroup>

    <ItemGroup>
        <ProjectReference Include="..\..\..\Partas.Solid\Partas.Solid.fsproj" />
    </ItemGroup>

</Project>
