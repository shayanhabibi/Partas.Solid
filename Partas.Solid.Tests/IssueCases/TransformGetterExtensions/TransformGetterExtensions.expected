import { For, Show, splitProps } from "solid-js";
import { flexRender, createSolidTable } from "@tanstack/solid-table";
import { Types_TableOptions$1 } from "./Program.fs.jsx";

export function TransformGetterExtensions(props) {
    const [PARTAS_LOCAL, PARTAS_OTHERS] = splitProps(props, []);
    const table = createSolidTable({});
    return <div>
        <Show when={table.getRowModel().rows.length}
            fallback={<div />}>
            <For each={table.getRowModel().rows}>
                {(row, index) => <div data-state={row.getIsSelected() && "selected"}>
                    <For each={row.getVisibleCells()}>
                        {(cell, index_1) => <div>
                            {flexRender(...[cell.column.columnDef.cell, cell.getContext])}
                        </div>}
                    </For>
                </div>}
            </For>
        </Show>
    </div>;
}

